algo/rainforest.c

```diff
@@ -555,7 +555,7 @@ static inline uint32_t rf_rambox(rf256_ctx_t *ctx, uint64_t old) {
 }
 
 // write (_x_,_y_) at cell _cell_ for offset _ofs_
-static inline void rf_w128(uint64_t *cell, ulong ofs, uint64_t x, uint64_t y) {
+static inline void rf_w128(uint64_t *cell, u_long ofs, uint64_t x, uint64_t y) {
```

Comment:
could you try with uint64_t and put "l" (little L) instead of "I" (big I) on line 564

Reply:
yup
tested on linux, linux cross-compile for windows and msvc